pkg/controllers/machine_controller.go

```diff
@@ -154,12 +169,53 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 		}
 	}()
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Comment:
don't use same name for context in anonymous function. because is extremely hard know on first look which context you use, because you can use closure principle too.

Reply:
anonymous function can be defined as standard separate function.

Reply:
Do you need such complex logic? Technically you calculate differences between volume sizes in every reconcile loop anyway. So why we cannot reconcile machine every 1min or 5min. How many VM do you want to run at the same time? Because our current limit is 128 vm per host

Reply:
Here we do only enqueue if the machine needs to be reconciled. It's an anti-pattern to just enqueue objects if you like to achieve a trigger to act on the object.

Reply:
okay and what another 2 issues?

Reply:
i created new issues for them